webapp/src/ts/modals/edit-user/edit-user.component.ts

```diff
@@ -1,3 +1,54 @@
+import {MmModalAbstract} from "@mm-modals/mm-modal/mm-modal";
+import {Injectable} from "@angular/core";
+import {UserSettingsService} from "@mm-services/user-settings.service";
+import {BsModalRef} from "ngx-bootstrap/modal";
+
+
+Injectable({
+  providedIn: 'root'
+})
+export class EditUserAbstract extends MmModalAbstract {
+
+  editUserModel;
+
+  constructor(
+    private bsModalRef: BsModalRef,
+    private userSettingsService: UserSettingsService,
+  ) {
+    super();
+    this.determineEditUserModel()
+      .then(model => {
+        this.editUserModel = model;
+      })
+      .catch(err => {
+        console.error('Error determining user model', err);
+      });
+  }
+
+  cancel() {
+    this.bsModalRef.hide();
+  }
+
+  determineEditUserModel() {
+    return this.userSettingsService.get()
+      .then((user: any) => {
+        if (user) {
+          return {
+            id: user._id,
+            username: user.name,
+            fullname: user.fullname,
+            email: user.email,
+            phone: user.phone,
+            language: { code: user.language }
+          };
+        } else {
+          return {};
+        }
+      });
+  }
+}
+
+/*
 const passwordTester = require('simple-password-tester');
```

Comment:
Have you finished migrating this file? 
If yes, please remove the old commented code.

Reply:
Done.